packages/brokeneck-react/src/components/User.test.js

```diff
@@ -0,0 +1,210 @@
+        UserStatus: 'FORCE_CHANGE_PASSWORD',
+        UserCreateDate: '2021-01-07T12:33:07.571Z',
+        UserLastModifiedDate: '2021-01-07T12:33:07.571Z',
+        groups: [
+          {
+            GroupName: 'AnotherGroup',
+            Description: 'Another group description',
+            CreationDate: '2021-01-05T10:42:20.733Z',
+            LastModifiedDate: '2021-01-05T10:42:20.733Z',
+            users: { data: [], nextPage: '234567' }
+          }
+        ]
+      },
+      groups: { data: [], nextPage: '234567' }
+    }
+  }
+
+  if (/query LoadGroups\(/.test(query)) {
+    data = {
+      groups: { data: [], nextPage: '234567' }
+    }
+  }
+
+  if (/query LoadGroup\(/.test(query)) {
+    data = {
+      group: {
+        GroupName: 'AnotherGroup',
+        Description: 'Another group description',
+        CreationDate: '2021-01-05T10:42:20.733Z',
+        LastModifiedDate: '2021-01-05T10:42:20.733Z',
+        users: { data: [], nextPage: '234567' }
+      },
+      users: { data: [], nextPage: '234567' }
+    }
+  }
+
+  return deepmerge({ data, loading: false }, overrides)
+}
+
+const mockMutation = jest.fn()
+
+jest.mock('graphql-hooks', () => {
+  const originalLib = jest.requireActual('graphql-hooks')
+
+  return {
+    ...originalLib,
+    useMutation: jest.fn().mockImplementation(() => [mockMutation]),
+    useQuery: jest.fn()
+  }
+})
+
+const mockGoBack = jest.fn()
+
+jest.mock('react-router-dom', () => {
+  const originalLib = jest.requireActual('react-router-dom')
+
+  return {
+    ...originalLib,
+    useHistory: jest.fn().mockImplementation(() => ({ goBack: mockGoBack })),
+    useQuery: jest.fn()
+  }
+})
+
+afterEach(() => {
+  jest.clearAllMocks()
+})
+
+describe('user', () => {
+  it('should render user component', () => {
+    useQuery.mockImplementation(mockUseQuery())
+
+    render(withProviders(<User userId="1234" />), {
+      wrapper: RootContextWrapper
+    })
+
+    expect(screen.getByText(/Add to group/i)).toBeInTheDocument()
+    expect(screen.getByText(/Delete user/i)).toBeInTheDocument()
+    expect(screen.getAllByTestId(/user-group-chip/i).length).toEqual(1)
+  })
+
+  it('should open delete dialog when "Delete user" clicked', async () => {
+    useQuery.mockImplementation(mockUseQuery())
+
+    render(withProviders(<User userId="1234" />), {
+      wrapper: RootContextWrapper
+    })
+    screen.getAllByRole('button', { name: /Delete user/i })[0].click()
+
+    expect(screen.getByTestId('dialog-form')).toBeInTheDocument()
+    expect(screen.getByRole('button', { name: 'Confirm' })).toBeInTheDocument()
+  })
+
+  it('should delete user when "Delete user" dialog submitted', async () => {
+    useQuery.mockImplementation(mockUseQuery())
+
+    render(withProviders(<User userId="1234" />), {
+      wrapper: RootContextWrapper
+    })
+    screen.getAllByRole('button', { name: /Delete user/i })[0].click()
+
+    fireEvent.submit(screen.getByTestId('dialog-form'))
+
+    await waitFor(() =>
```

Comment:
do we need to wait for this to happen? I am not sure but I thought the mutation would be invoked right away after confirming the dialog, hence that it wouldn't be necessary to wrap this in waitFor. but I guess there may be a reason why you used it

Reply:
The wairFor is required because otherwise the promise the promise form the dialog box awaiting the submit isnt fully resolved, meaning that the mutation function is not fired by the time the tests get here. Not having the wait for causes them to fail. as a result of the function call not yet occuring